includes/Modules/Analytics_4.php

```diff
@@ -2139,4 +2164,136 @@ private function filter_analytics_allow_tracking_disabled( $allowed ) {
 
 		return $allowed;
 	}
+
+	/**
+	 * Sets and returns available audiences.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param GoogleAnalyticsAdminV1alphaAudience[] $audiences The audiences to set.
+	 * @return array The available audiences.
+	 */
+	private function set_available_audiences( $audiences ) {
+		$available_audiences = array_map(
+			function( GoogleAnalyticsAdminV1alphaAudience $audience ) {
+				$audience_item = array(
+					'name'        => $audience->getName(),
+					'displayName' => $audience->getDisplayName(),
+					'description' => $audience->getDescription(),
+				);
+
+				$audience_slug = $this->get_audience_slug( $audience );
+				$audience_type = $this->get_audience_type( $audience_slug );
+
+				$audience_item['audienceType'] = $audience_type;
+				$audience_item['audienceSlug'] = $audience_slug;
+
+				return $audience_item;
+			},
+			$audiences
+		);
+
+		$this->get_settings()->merge(
+			array(
+				'availableAudiences'             => $available_audiences,
+				'availableAudiencesLastSyncedAt' => time(),
+			)
+		);
+
+		return $available_audiences;
+	}
+
+	/**
+	 * Gets the audience slug.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param GoogleAnalyticsAdminV1alphaAudience $audience The audience object.
+	 * @return string The audience slug.
+	 */
+	private function get_audience_slug( GoogleAnalyticsAdminV1alphaAudience $audience ) {
+		$display_name = $audience->getDisplayName();
+
+		if ( 'All Users' === $display_name ) {
+			return 'all-users';
+		}
+
+		if ( 'Purchasers' === $display_name ) {
+			return 'purchasers';
+		}
+
+		$filter_clauses = $audience->getFilterClauses();
+
+		if ( ! $filter_clauses ) {
+			return '';
+		}
+
+		if ( $this->has_deep_value(
+			$filter_clauses,
+			'created_by_googlesitekit:new_visitors'
+		) ) {
+			return 'new-visitors';
+		}
+
+		if ( $this->has_deep_value(
+			$filter_clauses,
+			'created_by_googlesitekit:returning_visitors'
+		) ) {
+			return 'returning-visitors';
+		}
+
+		// Return an empty string for user defined audiences.
+		return '';
```

Comment:
I'm returning an empty string from user-defined audiences here so that we don't run into a missing field error.